src/project.cc

```diff
@@ -0,0 +1,556 @@
+#include "project.h"
+#include "config.h"
+#include "terminal.h"
+#include "filesystem.h"
+#include <fstream>
+#include "menu.h"
+#include "notebook.h"
+#ifdef JUCI_ENABLE_DEBUG
+#include "debug_clang.h"
+#endif
+
+std::unordered_map<std::string, std::string> Project::run_arguments;
+std::unordered_map<std::string, std::string> Project::debug_run_arguments;
+std::atomic<bool> Project::compiling;
+std::atomic<bool> Project::debugging;
+std::pair<boost::filesystem::path, std::pair<int, int> > Project::debug_stop;
+boost::filesystem::path Project::debug_last_stop_file_path;
+
+std::unique_ptr<Project::Language> Project::current_language;
+
+void Project::debug_update_status(const std::string &debug_status) {
+  if(debug_status.empty()) {
+    debug_status_label().set_text("");
+    auto &menu=Menu::get();
+    menu.actions["debug_stop"]->set_enabled(false);
+    menu.actions["debug_kill"]->set_enabled(false);
+    menu.actions["debug_step_over"]->set_enabled(false);
+    menu.actions["debug_step_into"]->set_enabled(false);
+    menu.actions["debug_step_out"]->set_enabled(false);
+    menu.actions["debug_backtrace"]->set_enabled(false);
+    menu.actions["debug_show_variables"]->set_enabled(false);
+    menu.actions["debug_run_command"]->set_enabled(false);
+    menu.actions["debug_goto_stop"]->set_enabled(false);
+  }
+  else {
+    debug_status_label().set_text("debug: "+debug_status);
+    auto &menu=Menu::get();
+    menu.actions["debug_stop"]->set_enabled();
+    menu.actions["debug_kill"]->set_enabled();
+    menu.actions["debug_step_over"]->set_enabled();
+    menu.actions["debug_step_into"]->set_enabled();
+    menu.actions["debug_step_out"]->set_enabled();
+    menu.actions["debug_backtrace"]->set_enabled();
+    menu.actions["debug_show_variables"]->set_enabled();
+    menu.actions["debug_run_command"]->set_enabled();
+    menu.actions["debug_goto_stop"]->set_enabled();
+  }
```

Comment:
I don't see the need of duplicating this.
menu.actions["debug_goto_stop"]->set_enabled(!debug_status.empty());